pytorch_lightning/trainer/initialization_mixin.py

```diff
@@ -0,0 +1,297 @@
+import os
+
+import torch
+
+from pytorch_lightning import _logger as log
+from pytorch_lightning.profiler import SimpleProfiler, PassThroughProfiler, BaseProfiler
+from pytorch_lightning.trainer.lightning_config import LightningConfig
+from pytorch_lightning.trainer.supporters import TensorRunningAccum
+from pytorch_lightning.trainer.distrib_parts import parse_gpu_ids, determine_root_gpu_device, pick_multiple_gpus
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities import rank_zero_warn
+
+try:
+    from apex import amp
+except ImportError:
+    APEX_AVAILABLE = False
+else:
+    APEX_AVAILABLE = True
+
+
+class TrainerInitializationMixin:
+    def init_setup(self, **config_kwargs):
+        r"""
+        Used to setup trainer and testing mixins from a kwargs argument that can be built into a
+        LightningConfig.
+
+        """
+        config = LightningConfig(**config_kwargs)
+
+        self.deterministic = config.deterministic
+        torch.backends.cudnn.deterministic = self.deterministic
+        if self.deterministic:
+            # fixing non-deterministic part of horovod
+            # https://github.com/PyTorchLightning/pytorch-lightning/pull/1572/files#r420279383
+            os.environ["HOROVOD_FUSION_THRESHOLD"] = str(0)
+
+        # Init callbacks
+        self.callbacks = config.callbacks or []
+        self.on_init_start()
+
+        # benchmarking
+        self.benchmark = config.benchmark
+        torch.backends.cudnn.benchmark = self.benchmark
+
+        # Transfer params
+        self.num_nodes = config.num_nodes
+        # Backward compatibility, TODO: remove in v0.8.0
+        if config.nb_gpu_nodes is not None:
+            rank_zero_warn(
+                "Argument `nb_gpu_nodes` has renamed to `num_nodes` since v0.5.0"
+                " and this method will be removed in v0.8.0",
+                DeprecationWarning,
+            )
```

Comment:
if we have now the `LightningConfig` this warning and deprecation shall be called there...